Message queue not empty at test start up

I don't know if this is a bug or a feature but without explicitly saying it, the message queue is at "module" scope (or even more global) which when using pytest (for example) is really annoying.

I'll illustrate it with the example from [dpytest doc](https://dpytest.readthedocs.io/en/latest/tutorials/using_pytest.html#starting-with-pytest) (only slightly modified: I renamed the `bot` fixture to `config_bot`: a weird habit of mine to always rename fixture object in test, feels like functions calls I guess :man_shrugging: )

```py
import discord
import discord.ext.commands as commands
from discord.ext.commands import Cog, command
import pytest
import pytest_asyncio
import discord.ext.test as dpytest


class Misc(Cog):
    @command()
    async def ping(self, ctx):
        await ctx.send("Pong !")

    @command()
    async def echo(self, ctx, text: str):
        await ctx.send(text)

@pytest_asyncio.fixture
async def config_bot():
    intents = discord.Intents.default()
    intents.members = True
    intents.message_content = True
    b = commands.Bot(command_prefix="!",
                     intents=intents)
    await b._async_setup_hook()  # setup the loop
    await b.add_cog(Misc())

    dpytest.configure(b)
    return b

@pytest.mark.asyncio
async def test_ping(config_bot):
    bot = config_bot
    await dpytest.message("!ping")
    assert dpytest.verify().message().content("Pong !")


@pytest.mark.asyncio
async def test_echo(config_bot):
    bot = config_bot
    await dpytest.message("!echo Hello world")
    assert dpytest.verify().message().contains().content("Hello")
```

as expected this example pass:
```
$ pytest
======================================= test session starts =======================================
platform linux -- Python 3.10.10, pytest-7.2.2, pluggy-1.0.0
rootdir: /tmp/dpytest
plugins: asyncio-0.21.0
asyncio: mode=strict
collected 2 items

test_dpytest_bug.py .. 

======================================== 2 passed in 0.24s ========================================
```

However just modifying a bit `test_ping` (sending two commands) will **crash `test_echo`** which seems weird: a test shouldn't influence another one:
```py
@pytest.mark.asyncio
async def test_ping(config_bot):
    bot = config_bot
    await dpytest.message("!ping")
    await dpytest.message("!ping")
    assert dpytest.verify().message().content("Pong !")
```

```
$ pytest
======================================= test session starts =======================================
[blabla]

test_dpytest_bug.py .F                                                                      [100%]

============================================ FAILURES =============================================
____________________________________________ test_echo ____________________________________________

config_bot = <discord.ext.commands.bot.Bot object at 0x7f9244d70490>

    @pytest.mark.asyncio
    async def test_echo(config_bot):
          [blablabla]

test_dpytest_bug.py:43: AssertionError
===================================== short test summary info =====================================
FAILED test_dpytest_bug.py::test_echo - AssertionError: assert <discord.ext.test.verify.VerifyMessage object at 0x7f9244d72200>
=================================== 1 failed, 1 passed in 0.34s ===================================
```

because the AssertionError dump isn't really clear (related to #115) I'll use `dpytest.get_message()` to debug my `test_echo` (the one which has crashed):
```py
@pytest.mark.asyncio
async def test_echo(config_bot):
    bot = config_bot
    await dpytest.message("!echo Hello world")
    print(dpytest.get_message().content)
    assert False
```

and we get:
```
$ pytest
======================================= test session starts =======================================
[blabla]

test_dpytest_bug.py .F                                                                      [100%]

============================================ FAILURES =============================================
____________________________________________ test_echo ____________________________________________
[blabla]
-------------------------------------- Captured stdout call ---------------------------------------
Pong !
===================================== short test summary info =====================================
FAILED test_dpytest_bug.py::test_echo - assert False
=================================== 1 failed, 1 passed in 0.34s ===================================_
```

Here we are: 
**in `test_echo` that doesn't call `!ping` tehre is a "Pong !" response in message queue** 

I'm aware of the `dpytest.empty_queue` to clear the message queue, but I'd expect the configuration from dpytest to be fucntion scoped. 
Ideally I'd like to call this `dpytest.empty_queue` in test teardown: for example after yielding `b` in the `config_bot` fixture but for some reason an async fixture can't yield and then teardown the test properly, at least I didn't manage to do it :confused: 

I'd really like your thoughts and help on that subject!

--- a/docs/tutorials/getting_started.rst
+++ b/docs/tutorials/getting_started.rst
@@ -44,6 +44,10 @@ it goes into the ``discord.py`` extensions module. So, the most basic usage of d
     asyncio.run(test_ping())
     asyncio.run(test_foo())
 
+One problem that could happen is that the ``sent_queue`` is shared between the tests. So in order not to mess between
+your tests (``verify()`` pops **one** message from the queue, so in general, you won't need to do anything) you can
+explicitly call ``empty_queue()``, as shown in the next example (and later, in the ``conftests.py``).
+
 If that looks like a lot of code just to run tests, don't worry, there's a better way! We can use pytest,
 a popular Python testing library.
 

--- a/docs/tutorials/getting_started.rst
+++ b/docs/tutorials/getting_started.rst
@@ -32,6 +32,7 @@ it goes into the ``discord.py`` extensions module. So, the most basic usage of d
         dpytest.configure(bot)
         await dpytest.message("!ping")
         assert dpytest.verify().message().contains().content("Ping:")
+        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     async def test_foo():
@@ -39,6 +40,7 @@ it goes into the ``discord.py`` extensions module. So, the most basic usage of d
         dpytest.configure(bot)
         await dpytest.message("!hello")
         assert dpytest.verify().message().content("Hello World!")
+        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     asyncio.run(test_ping())

--- a/docs/tutorials/getting_started.rst
+++ b/docs/tutorials/getting_started.rst
@@ -32,7 +32,6 @@ it goes into the ``discord.py`` extensions module. So, the most basic usage of d
         dpytest.configure(bot)
         await dpytest.message("!ping")
         assert dpytest.verify().message().contains().content("Ping:")
-        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     async def test_foo():
@@ -40,7 +39,6 @@ it goes into the ``discord.py`` extensions module. So, the most basic usage of d
         dpytest.configure(bot)
         await dpytest.message("!hello")
         assert dpytest.verify().message().content("Hello World!")
-        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     asyncio.run(test_ping())

--- a/docs/tutorials/using_pytest.rst
+++ b/docs/tutorials/using_pytest.rst
@@ -17,8 +17,9 @@ as:
 
 ``pytest`` will detect any functions starting with 'test' in directories it searches, and run them. It also supports
 a feature we will use heavily, called 'fixtures'. Fixtures are functions that do some common test setup, and
-then can be used in tests to always perform that setup. They can also return an object that will be passed to
-the test.
+then can be used in tests to always perform that setup, they can also return an object that will be passed to
+the test. Finally, they allow you to perform test teardown, cleaning message queue for example.
+See https://docs.pytest.org/en/latest/how-to/fixtures.html#teardown-cleanup-aka-fixture-finalization
 
 The final piece of this is ``pytest-asyncio``, a library for allowing ``pytest`` to run async tests. It is
 automatically installed when you get ``dpytest`` from pip, so you don't need to worry about installing it.

--- a/docs/tutorials/using_pytest.rst
+++ b/docs/tutorials/using_pytest.rst
@@ -47,6 +47,7 @@ Putting all this together, we can rewrite our previous tests to look like this:
 
     @pytest_asyncio.fixture
     async def bot():
+        # Setup
         intents = discord.Intents.default()
         intents.members = True
         intents.message_content = True
@@ -56,7 +57,11 @@ Putting all this together, we can rewrite our previous tests to look like this:
         await b.add_cog(Misc())
 
         dpytest.configure(b)
-        return b
+
+        yield b
+
+        # Teardown
+        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     @pytest.mark.asyncio
@@ -99,6 +104,7 @@ An example ``conftest.py`` might look like this:
 
     @pytest_asyncio.fixture
     async def bot():
+        # Setup
         intents = discord.Intents.default()
         intents.members = True
         intents.message_content = True
@@ -106,12 +112,11 @@ An example ``conftest.py`` might look like this:
                         intents=intents)
         await b._async_setup_hook()
         dpytest.configure(b)
-        return b
 
+        yield b
 
-    @pytest_asyncio.fixture(autouse=True)
-    async def cleanup():
-        await dpytest.empty_queue()
+        # Teardown
+        await dpytest.empty_queue() # empty the global message queue as test teardown
 
 
     def pytest_sessionfinish(session, exitstatus):

--- a/docs/tutorials/using_pytest.rst
+++ b/docs/tutorials/using_pytest.rst
@@ -143,6 +143,12 @@ Troubleshooting
 Make sure your tests take a parameter with the exact same name as the fixture,
 pytest runs them based on name, including capitalization.
 
+- I use dpytest.verify().message() and it fails but it shouldn't
+- dpytest.get_message() returns a message from another test
+
+Make sure you properly emptied the queue in the previous test, otherwise you
+could have remaining messages from previous tests messing up.
+
 --------------------
 
 This is currently the end of the tutorials. Take a look at the `Runner Documentation`_ to see all the things you can

